### PR TITLE
Automatic update of searx/translations/messages.pot and messages.po files.

### PR DESCRIPTION
**Pull request by dalf**
## What does this PR do?

See commit messages:
* remove .tx directory (transifex)
* babel.cfg: ignore node_modules directories
* update searx_extra/update/update_translations.sh
   * the script expects searx/translations/messages.pot to be commited
   * it calls "pybabel extract", and if there is a meaningful change,
   * it calls "pybabel update"
* github workflow:
   * after each on the master branch of searxng (not elsewhere), a call `searx_extra/update/update_translations.sh` check if there is some changes on `searx/translations/messages.pot`
   * if yes it create a PR with a commit similar to 596577e907607b582709aa7afd51a0cc9df9ca9f so weblate can get the last messages.
* update the messages.po files (it haven't been done since a while).

The workflow is disabled on this branch, how it is on the master branch:
image

see https://github.com/dalf/searxng/actions/runs/1085458894

If this github workflow doesn't create issue later the PR can be replaced by a commit on the master branch.

## Why is this change important?

Related to #195 / integration of weblate.

## How to test this PR locally?

* call  `searx_extra/update/update_translation.sh`
* call again `searx_extra/update/update_translation.sh`: see there is no change (no call to `pybabel update`)
* change `searx/translations/messages.pot`
* call again `searx_extra/update/update_translation.sh`: check that the script calls `pybabel update`

The github workflow can only be tested on the master branch, add a commit to replace `searxng`:
https://github.com/searxng/searxng/blob/53156cd4c66c5fe0a6059515f5df7b7bbe02d35a/.github/workflows/integration.yml#L96

## Related issues

Related to #195

### CODE REVIEW

**Comment by dalf** [comment]
The github workflow works:
* it creates a PR when `searx/translations/messages.pot` change
* and nothing when it is not the change (and it doesn't fail).

I'm still in favor of a PR to update `searx/translations/messages.pot` for now.

@return42 : can you check https://github.com/searxng/searxng/pull/238/commits/016349aea34182aaef5d99d981d20d3c41431e54 ?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -90,6 +90,47 @@ jobs:
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+        architecture: 'x64'
+    - name: Update transations
+      id: update
+      continue-on-error: true
+      run: |
+        pip install babel jinja2
+        searx_extra/update/update_translations.sh
```
Comment (return42) [inline .github/workflows/integration.yml]: This code should be executed in the virtualenv (./local/py3) / not in the environment from the OS.

**Comment by return42** [comment]
@dalf thanks for cleaning up the translation tasks!  .. I placed 1a66cad on top of your branch (see commit message) .. could you please test the CI (git workflow) part?

File: babel.cfg [hunk 2]
```diff
@@ -1,3 +1,4 @@
+[ignore: **/node_modules/**]
```
Comment (return42) [inline babel.cfg:1]: See my https://github.com/searxng/searxng/issues/141#issuecomment-890481065

File: .github/workflows/integration.yml (hunk 1 above)
Comment (dalf) [reply .github/workflows/integration.yml]: Note: the CI runs this code in a throw away VM: everything runs as root.

Comment (return42) [reply .github/workflows/integration.yml]: Anyway, we do all build task in a dedicated environment and this is the environment in ./local/py3

By example: I tried to update translations and it fails on my desktop PC.

With commit 900baf2  all pybabel commands are executed in this environment by  the leading `pyenv.cmd pybabel`

**Review by return42 — APPROVED** [review]
LGTM .. lets merge.

**Comment by dalf** [comment]
Tested on dalf/searxng. It works as expected:
* https://github.com/dalf/searxng/actions/runs/1089591592 : update
* https://github.com/dalf/searxng/actions/runs/1089560765 : no updates

When there is no update, the [output](https://github.com/dalf/searxng/runs/3218846086?check_suite_focus=true) :
```
 extracting messages from searx/search/processors/online_dictionary.py
extracting messages from searx/shared/__init__.py
extracting messages from searx/shared/shared_abstract.py
BABEL     no changes detected, exiting
extracting messages from searx/shared/shared_simple.py
extracting messages from searx/shared/shared_uwsgi.py
extracting messages from searx/templates/oscar/404.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/oscar/about.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/oscar/advanced.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_"
...
extracting messages from searx/templates/simple/search.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/stats.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/time-range.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/messages/no_results.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/code.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/default.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/images.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/key-value.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/map.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/torrent.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
extracting messages from searx/templates/simple/result_templates/videos.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
writing PO template file to searx/translations/messages.pot
ERROR: ci.babel.update exit with error (1)
make: *** [Makefile:96: ci.babel.update] Error 1
Error: Process completed with exit code 2.
```

I don't understand why: `BABEL     no changes detected, exiting` is lost in the middle of `pybabel extract` output.

**Comment by return42** [comment]
I suppose it is about _flushing_ stdout and stderr pipes.  The `BABEL`is printed to stdout while `pybabel extract` logs on stderr.

I can't reproduce it on my (desktop) systems otherwise I could have a look ..

**Comment by dalf** [comment]
>  I could have a look ..

If the script remains as it is, it can be helpful to have the message at the end.

But there is the integration of weblate to add to ./manage:
`pybabel extract ; git commit` should not happen before [`wlc push`](https://docs.weblate.org/en/latest/wlc.html#cmdoption-wlc-arg-push) otherwise the weblate repository is stuck with a conflict.

**Pull request merged**